src/Npgsql/NpgsqlCommand.cs

```diff
@@ -1401,12 +1402,34 @@ void LogCommand()
                 var p = s.InputParameters;
                 if (p.Count > 0 && (NpgsqlLogManager.IsParameterLoggingEnabled || connector.Settings.LogParameters))
                 {
-                    sb.Append('\t').Append("Parameters:");
                     for (var i = 0; i < p.Count; i++)
-                        sb.Append("\t$").Append(i + 1).Append(": ").Append(Convert.ToString(p[i].Value, CultureInfo.InvariantCulture));
+                    {
+                        switch (p[i].Value)
+                        {
+                        case IList list:
+                            for (var j = 0; j < list.Count; j++)
+                            {
+                                if (j == 0)
+                                    LogParamHelper(sb, i + 1).Append(": ");
+                                sb.Append("\t#").Append(j).Append(": ").Append(Convert.ToString(list[j], CultureInfo.InvariantCulture));
+                            }
+                            break;
+                        case DBNull _:
+                        case null:
+                            LogParamHelper(sb, i + 1).Append(":\t").Append(Convert.ToString("null", CultureInfo.InvariantCulture));
+                            break;
+                        default:
+                            LogParamHelper(sb, i + 1).Append(":\t").Append(Convert.ToString(p[i].Value, CultureInfo.InvariantCulture));
+                            break;
+                        }
+                        sb.AppendLine();
+                    }
                 }
             }
-
+            StringBuilder LogParamHelper(StringBuilder sb, int idx)
```

Comment:
nit: can be an expression-bodied method (but see above, probably no need for an internal method here at all)